src/kudu/consensus/raft_consensus.cc

```diff
@@ -2678,6 +2684,19 @@ void RaftConsensus::DoElectionCallback(ElectionReason reason, const ElectionResu
   }
 }
 
+void RaftConsensus::NestedElectionDecisionCallback(
+    ElectionReason reason, const ElectionResult& result) {
+  DoElectionCallback(reason, result);
+  const bool was_pre_election = result.vote_request.is_pre_election();
+  UniqueLock lock(lock_);
+  const bool runClientElectionDecisionCallback =
+      (cmeta_->active_role() == RaftPeerPB::LEADER) || !was_pre_election;
```

Comment:
Is this what you intended as the logic?
it seems for true elections, you will get notifications of failure.
For pre-elections, you will get notifications if you are the leader.

I thought the logic should be && condition
or just plain !was_pre_election (notify both success and failures), in which you case
you wont need the extra locking.